AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLogging.kt

```diff
@@ -0,0 +1,101 @@
+package com.automattic.android.tracks.crashlogging
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.automattic.android.tracks.crashlogging.internal.SentryErrorTrackerWrapper
+import io.sentry.SentryEvent
+import io.sentry.SentryLevel
+import io.sentry.SentryOptions
+import io.sentry.protocol.Message
+import io.sentry.protocol.User
+
+object CrashLogging {
+
+    private lateinit var dataProvider: CrashLoggingDataProvider
+    private lateinit var sentryWrapper: SentryErrorTrackerWrapper
+
+    @JvmStatic
+    fun start(
+        context: Context,
+        dataProvider: CrashLoggingDataProvider,
+    ) {
+        start(context, dataProvider, SentryErrorTrackerWrapper())
+    }
+
+    @VisibleForTesting
+    internal fun start(
+        context: Context,
+        dataProvider: CrashLoggingDataProvider,
+        sentryWrapper: SentryErrorTrackerWrapper
+    ) {
+        this.sentryWrapper = sentryWrapper
+        this.dataProvider = dataProvider
+
+        sentryWrapper.initialize(context) { options ->
+            options.apply {
+                dsn = dataProvider.sentryDSN
+                environment = dataProvider.buildType
+                release = dataProvider.releaseName
+                setDebug(dataProvider.enableCrashLoggingLogs)
```

Comment:
Setting `BuildConfig.DEBUG`, which I changed from in 58e2187e3a10cebf66665c8c3ad7d0b19fb77a5e, did not make sense as I forgot that this is not project-included library, so `BuildType` of project won't match `BuildType` of library. We release `release` variant on Jitpack so we would always had `setDebug(false)` in clients.

I've updated `CrashLoggingDataProvider` to provide wether client wants to see logs or not.

Reply:
Ohh, great catch! 👏